3_visualize/src/interactive_map.R

```diff
@@ -0,0 +1,56 @@
+# Build interactive map with leaflet
+build_map_leaflet <- function(p3_huc8_sf, p3_saline_lakes_sf, p3_flowlines_sf, p3_gage_sites_sf){
+  # Define color ramp for steams
+  pal_stream <- colorFactor(
+    palette = scico::scico(9, palette = 'davos')[6:3],
+    domain = p3_flowlines_sf$streamorde_size)
+  
+  # Define color ramp for gages
+  pal_gage <- colorFactor(
+    palette = scico::scico(16, palette = 'davos')[c(11, 1)],
+    domain = p3_gage_sites_sf$in_HUC8)
+  
+  # Build map
+  leaflet() %>% 
+    #Add Basemap
+    addProviderTiles("Esri.WorldGrayCanvas", group = c("No labels", "Labels"))  %>%
+    addProviderTiles("CartoDB.PositronOnlyLabels", group = "Labels") %>%
+    
+    # Add data layers
+    addPolygons(data = p3_huc8_sf, group = "Subbasin (HUC 8)",
+                color = "#C3CB9F", opacity = 0.4, weight = 3,
+                popup = ~label) %>% 
+    addPolygons(data = p3_saline_lakes_sf, group = "Saline lakes",
+                color = '#3A6DB7', opacity = 0.8, weight = 2,
+                popup = ~label) %>%
+    addPolylines(data = p3_flowlines_sf, group = "Streams",
+                 color = ~pal_stream(streamorde), opacity = 0.8,
+                 weight = ~streamorde,
+                 popup = ~label) %>%
```

Comment:
To keep things simpler, I'd use a single color for the flowlines but still use streamorder for size.